### PR TITLE
Week4

### PR DESCRIPTION
**Pull request by KhawYewOnn**
Complete requirement for week 4

### CODE REVIEW

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,120 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
```
Comment (rashikaranpuria) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:17]: Let's avoid folded imports

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,120 @@
...
 
+/*
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```
Comment (rashikaranpuria) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:21]: Remove stray comments

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,120 @@
...
     throw new UnsupportedOperationException("TODO: Implement this method.");
   }
-}
+}*/
+
+
```
Comment (rashikaranpuria) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:26]: Extra blankspace

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -14,10 +14,120 @@
...
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public ArrayList<TimeRange> filterTimes(Collection<Event> events, HashSet<String> attendees) {
+    ArrayList<TimeRange> times = new ArrayList<TimeRange>();
+    for (Event event: events) {
+      HashSet<String> curr_attendees = new HashSet<String>();
```
Comment (rashikaranpuria) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:41]: Use camelCase naming to keep it similar to rest of the variable names

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (rashikaranpuria) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:26]: extra newline

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -14,10 +14,120 @@
...
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     throw new UnsupportedOperationException("TODO: Implement this method.");
   }
-}
+}*/
+
+
+public final class FindMeetingQuery {
+  public boolean hasMatch(HashSet<String> first, HashSet<String> second) {
```
Comment (rashikaranpuria) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:28]: first and second as argument names sound unclear as to what they hold

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (KhawYewOnn) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:17]: Noted, I have changed it to 
```
import java.util.HashSet;
import java.util.Iterator;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Collection;
```

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (KhawYewOnn) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:21]: okay

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (KhawYewOnn) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:26]: okay removed it

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 4 above)
Comment (KhawYewOnn) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:41]: okay, changed `curr_attendees` to `currAttendees`

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 5 above)
Comment (KhawYewOnn) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:28]: Okay, what should I change it to? I saw on `Event.java` file that they used `a` and `b` for this kind of compare function. Should I change it to `a` and `b`?

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 6]
```diff
@@ -15,9 +15,118 @@
...
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
+  /**
+  Check if any of the string that appear in stringSet appear in otherSet
```
Comment (rashikaranpuria) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: Move comment to same line as opening and closing comment of the comment

**Pull request merged**